aggregation/quantile/cm/stream.go

```diff
@@ -311,7 +345,9 @@ func (s *stream) addToMinHeap(heap *minHeap, value float64) {
 	// and return the current heap to the pool
 	if len(curr) == cap(curr) {
 		newHeap := s.floatsPool.Get(2 * len(curr))
-		newHeap = append(newHeap, curr...)
+		for i := 0; i < len(curr); i++ {
```

Comment:
Aggh, yea, if the slice we get back has zero length then `copy` won't work. Any reason to not use `append(newHeap, curr...)` then? It seems like the compiler should be able to optimize that into a single copy under the hood.